runtime/queries/cpp/highlights.scm

```diff
@@ -15,14 +15,11 @@
   declarator: (_) @variable.parameter)
 
 ;(field_expression) @variable.parameter ;; How to highlight this?
-(((field_expression
-  (field_identifier) @function.method))@_parent(#has-parent? @_parent template_method function_declarator))
+(((field_expression (field_identifier) @function.method))@_parent(#has-parent? @_parent template_method function_declarator))
```

Comment:
Here's another instance where the predicate gets smushed together with the captures (not new in this commit, just noticed).

Reply:
There's a number of `)@` patterns in the queries introduced by the formatter?

Reply:
I just didn't expect people to do captures after a grouping :bow: 

fyi, `((foo) (bar)) @capture` is equivalent to `(foo) @capture` so I'm unsure why people write it like that

Reply:
Fixed with latest force push

Reply:
People have no clue and just spam parens until it works. (At least, that's what I do.)

Is this something we can normalize in the formatter (always replace the former by the latter)? (Not worth complicating the script significantly, but would be nice to have if it's simple).

Reply:
It's too much. The formatter will also need to do refactoring work.

Formatters should not be allowed to move nodes around that change how the syntax tree works like that tbh, that's a linters/others jobs

Reply:
(With that said). It can help remove groupings (paren spamming). But I will rather look at it later with list `[]` removals.

Reply:
That makes sense, yes.

Reply:
Still not going to look at grouping stuffs. I cannot truly confirm the behavior for this.